internals/overlord/servstate/handlers.go

```diff
@@ -340,16 +357,34 @@ func (s *serviceData) startInternal() error {
 	s.cmd = exec.Command(args[0], args[1:]...)
 	s.cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
 
-	// Copy environment to avoid updating original.
 	environment := make(map[string]string)
-	for k, v := range s.config.Environment {
-		environment[k] = v
+	if s.workload != nil {
+		maps.Copy(environment, s.workload.Environment)
 	}
+	maps.Copy(environment, s.config.Environment)
 
 	s.cmd.Dir = s.config.WorkingDir
 
 	// Start as another user if specified in plan.
-	uid, gid, err := osutil.NormalizeUidGid(s.config.UserID, s.config.GroupID, s.config.User, s.config.Group)
+	var uid, gid *int
+	var username, groupname string
+	if s.workload != nil {
```

Comment:
As per our previous conversation:

```
[Ben's notes in the comment below] it would be a validation error to specify user/user-id/group/group-id on the service (in the combined plan) if a workload is specified
```

Is this true in the implementation ? The code looks like service uid/gid would replace workloads. Is it prevented at a validation phase ?